### ISSUE TITLE
Error when calling `socket.getaddrinfo`

### ISSUE THREAD

**Issue by bjmb** [eventlet/eventlet#439]
```python
from eventlet import monkey_patch
monkey_patch()
import socket

ip = "1"

info = socket.getaddrinfo(ip, 9042, socket.AF_UNSPEC, socket.SOCK_STREAM)
print("info: {}".format(info))
```
The previous code fails with `socket.gaierror(socket.EAI_NONAME, 'No address found')`. But if the two first lines are corrected the output is `info: [(2, 1, 6, '', ('0.0.0.1', 9042))]`

**Comment by temoto** [comment]
@bjmb given `ip="1"` both invalid hostname and address, error seems more appropriate, right?

**Comment by bjmb** [comment]
Agreed, but shouldn't the behavior be the same with and without patching?

**Comment by bjmb** [comment]
Agreed it shouldn't be the same, but the returned value of the python library methods should be the same: `Eventlet is a concurrent networking library for Python that allows you to change how you run your code, not how you write it.`, if I get a different return with and without eventlet it's going to change how I write my code.
I'm not saying an error shouldn't be risen, maybe the python implementation should be different.

**Comment by temoto** [comment]
Yep, so we both aren't sure.

What could lead to certainty is looking at patched/original behavior for a valid input. Do you get any unexpected results with proper inputs for getaddrinfo?

**Comment by bjmb** [comment]
I get the same expected behavior for both with a proper input: `info: [(2, 1, 6, '', ('127.0.0.1', 9042))]`
